tests/utils.py

```diff
@@ -61,6 +73,23 @@ def get_mypy_req() -> str:
         return next(strip_comments(line) for line in requirements_file if "mypy" in line)
 
 
+# ====================================================================
+# Parsing the requirements file
+# ====================================================================
+
+
+def parse_versions_from_requirements() -> dict[str, str]:
+    """Return the requested version for each package in the requirements file."""
+    reqs: dict[str, str] = {}
+    for line in open("requirements-tests.txt"):
+        pkg_and_version = line.split("#")[0].split(";")[0]
+        if "==" not in pkg_and_version:
+            continue
+        package, version = pkg_and_version.split("==", 1)
+        reqs[package.strip()] = version.strip()
+    return reqs
```

Comment:
Feels quite similar to the `get_mypy_req()` function immediately above.

I'd consider just doing

```py
with open("requirements-tests.txt", encoding="utf-8") as f:
    pyright_req = next(strip_comments(line) for line in f if "pyright==" in line)
```

inside `pyright_test.py` for now, unless we know of other places that need the pyright version.

Reply:
I've instead opted to replace get_mypy_req() with this more robust function

Reply:
Won't work, since we only need the version. But still I prefer the more robust parsing here.